packages/@aws-cdk/aws-stepfunctions/lib/state-machine.ts

```diff
@@ -118,6 +118,13 @@ export interface StateMachineProps {
    * @default No logging
    */
   readonly logs?: LogOptions;
+
+  /**
+   * Specifies whether Amazon X-Ray tracing is enabled for this state machine.
+   *
+   * @default false
+   */
+  readonly tracingEnabled?: boolean;
```

Comment:
since an update does not require an [interruption](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-stepfunctions-statemachine.html#cfn-stepfunctions-statemachine-tracingconfiguration), I'm thinking the default should be to enable tracing. 

are there any compelling reasons why this should be disabled by default?

Reply:
So, in this implementation, I am trying to stay consistent with the xray support in both lambda ([cfn](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-lambda-function.html#cfn-lambda-function-tracingconfig), [cdk](https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-lambda/lib/function.ts#L192)) and apig ([cfn](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-apigateway-stage.html#cfn-apigateway-stage-tracingenabled), [cdk](https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-apigateway/lib/stage.ts#L54)).

For lambda and apig, the cfn resources have tracing disabled by default and the constructs seems to have followed that pattern so I think we should be doing the same here which makes us consistent with both cfn defaults and the pattern that already exists in cdk

Reply:
gotcha, I think that's a reasonable approach and default in that case. thanks for looking into those